cfgov/v1/routing.py

```diff
@@ -46,3 +53,32 @@ def get_page_relative_url(page, site):
         return page_path
     else:
         return root_url + page_path
+
+
+@register.assignment_tag(takes_context=True)
+def get_page_state_url(context, page):
+    page = Page.objects.get(id=page.id).specific
+    if not page.live and not page.shared:
+        return None
+    url = page.url
+    if url is None:  # If page is not aligned to a site root return None
+        return None
+    page_hostname = urlparse(url).hostname
+    staging_hostname = os.environ.get('DJANGO_STAGING_HOSTNAME')
```

Comment:
Ah-- nm, it's a method that we stick on to the request via middleware, not useful here